### PR TITLE
LOGIC: user service - createUser method #6

### PR DESCRIPTION
**Pull request by alpha037**
### Implemented the following logic:
 -  [x] Username must be of a minimum of 5 characters.
bad-username

-  [x] Password must be of a minimum of 8 characters.
bad-password

- [x] Password must not be the word *password* itself.
password-itself

- *User creation successful (needed for the next step)*
user-created

- [x] Check if the username already exists.
user-already-exists

### CODE REVIEW

**Review by ohbus — APPROVED** [review]
LGTM

**Comment by alpha037** [comment]
@ohbus, thanks!

**Comment by alpha037** [comment]
- [x] Fix: Password cannot **contain** the word *password* itself.
password-itself

**Comment by championswimmer** [comment]
Eventually we should have a separate util for checking password strength 
But just a basic check on `password` check is ok for now.

File: src/users/users.service.ts [hunk 1]
```diff
@@ -33,10 +37,25 @@ export class UsersService {
...
+
+    const usernameAlreadyExists = await this.getUserByUsername(user.username);
+    if (usernameAlreadyExists)
+      throw new ConflictException('This username is already taken!');
+
     const newUser = await this.userRepo.save(user);
-    // TODO: check for username/email existing and throw proper error
-    // TODO: check username min length 5 chars
-    // TODO: check password min length 8 chars
+    // TODO: check for username/email existing and throw proper error - ✅
```
Comment (championswimmer) [inline src/users/users.service.ts]: remove the TODOs from here 

Ctrl-R `todo` is the best way to find out pending stuff in a project. The comments should not remain TODO if the work is DONE already.

Comment (alpha037) [reply src/users/users.service.ts]: Noted. I've removed the TODOs in `users.service.ts`.

**Pull request merged**